libchisel/src/dropsection.rs

```diff
@@ -0,0 +1,172 @@
+use super::{ModuleError, ModuleTranslator};
+
+use parity_wasm::builder::*;
+use parity_wasm::elements::*;
+
+/// Enum on which ModuleTranslator is implemented.
+pub enum DropSection<'a> {
+    NamesSection,
+    /// Name of the custom section.
+    CustomSectionByName(&'a String),
+    /// Index of the custom section.
+    CustomSectionByIndex(usize),
+    /// Index of the unknown section.
+    UnknownSectionByIndex(usize),
+}
+
+fn names_section_index_for(module: &Module) -> Option<usize> {
+    for (index, section) in module.sections().iter().enumerate() {
```

Comment:
You can iterate over it and use `position` to find the section in a more concise way. For example:
`module.sections().iter().position(|e| match e { Section::Name(_) => true, _ => false })`